mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -262,6 +262,22 @@ static TransformingForOp createLdsStoreLoop(OpBuilder &b, Location loc,
   return loop;
 }
 
+/// Temporary function to remove the kPack transformation currently being
+/// introduced in conv-to-gemm. kPack is a LDS-specific transformation and
+/// should not have been applied to the gridwise gemm argument. Future work will
+/// make the conversion stop doing that, and we'll be able to get rid of this.
+static Value unKpack(OpBuilder &b, Value matrix) {
+  ArrayRef<int64_t> shape = matrix.getType().cast<MemRefType>().getShape();
+  if (shape.size() != 4)
+    return matrix;
+  BottomUpTMBuilder unkpack(b, {"g", "k", "d", "kpack"}, shape,
+                            matrix.getLoc());
+  unkpack.passThrough({"g", "m"});
```

Comment:
... for some reason the comment I left didn't show up here, so yeah, spelling mistake and it's surprising that this doesn't explode.

Reply:
I can dig into this a bit. It is surprising how this can surprisingly hide another bug as you mentioned [here](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/768#discussion_r974620575).